### ISSUE TITLE
[Distributed Index] User not getting docs in  _changes when given access via sync function

### ISSUE THREAD

**Issue by sethrosetter** [couchbase/sync_gateway#1308]
Testcase: https://github.com/couchbaselabs/sync-gateway-testcluster/blob/sync-scenarios-1/functional_tests/test_sync.py

dj_0 user is created via Admin REST with no channels
100 docs are pushed to "KDWB" channel
A doc name "access_doc" is pushed

Using the sync functions below

```
function(doc, oldDoc){
    if(doc._id == "access_doc"){
        access("dj_0", ["KDWB"]);
    }
    channel(doc.channels);
}
```

After all the docs have been pushed, I issue a 4984:/db/_changes request for dj_0
### Expected:

```
I should see 100 docs returned
```
### Actual:

```
Couchbase Sync Gateway/0.0.1(360;b85b1c8) - I see 0 docs returned
Couchbase Sync Gateway/master(7fce059) - I see 100 docs returned
```

**Comment by adamcfraser** [comment]
Looks to me like this is pending https://github.com/couchbase/sync_gateway/issues/993, which in turn is waiting for the current perf optimizations to be completed.

**Comment by adamcfraser** [comment]
This should be fixed now that #993 is in merged.  

Closing - @sethrosetter can reopen if this test is still failing once 1.4 sg_accel testing spins up.
